### ISSUE TITLE
helm completion does not auto-complete releases or charts

### ISSUE THREAD

**Issue by thomas-riccardi** [helm/helm#3170]
helm completion does not auto-complete releases or charts:
```bash
# Releases
$ helm ls
NAME            REVISION        UPDATED                         STATUS          CHART           NAMESPACE
smelly-newt     3               Mon Nov 20 11:41:29 2017        DEPLOYED        rabbitmq-0.6.10 dev 
$ helm history <TAB>
# suggests local files instead of existing releases

# Charts
$ helm charts <TAB>
# suggests local files only, could also suggest charts from repositories
```

As an example, `kubectl` has auto-completion on remote resources, and it's really much more useful than what `helm` has.

**Comment by thomas-riccardi** [comment]
/remove-lifecycle stale

The feature is still missing as far as I know.

**Comment by Vickey-Wu** [comment]
```
helm completion bash
source <(helm completion bash)
```
[helm docs](https://docs.helm.sh/helm/#helm-completion)

**Comment by thomas-riccardi** [comment]
@Vickey-Wu yes there is a minimal bash auto-completion support in helm, but it does not dynamically auto-completes releases names and such like `kubectl` does.
I checked the new helm 2.10 version, as well as the changelog, commits and code: there is nothing new: the feature is still missing.

**Comment by bacongobbler** [comment]
Feel like working on a fix? :)

**Comment by cory-klein** [comment]
I've looked into how this is done in `kubectl` and it goes beyond my level of area expertise. I am 🤞  that somebody that has the matching cross section of skills between shell completion and the tools behind the `helm`/`kubectl` CLIs comes across this issue.

**Comment by marckhouzam** [comment]
The completion of releases names has been done in #5562 for v2 (but I don't think it has made it to a release yet) and #5829 for v3.

Still more dynamic completions can be added, such as charts, as suggested in this issue.

**Comment by marckhouzam** [comment]
The completion of charts has been implemented in #6730 for Helm 3.

The two dynamic completions requested in this issue will be part of helm 3 so I feel this issue can be closed.
cc @bacongobbler

**Comment by bacongobbler** [comment]
thanks @marckhouzam for the ping!
